### ISSUE TITLE
Linux: Autostart ownCloud client minimized

### ISSUE THREAD

**Issue by ldmpub** [owncloud/client#5528]
Hi,

Yet after each boot ownCloud client Window is displayed. It's annoying.

I searched the net and it seems to be a new feature request: to have under Linux (for me Xubuntu 16.04) the same behaviour (by default?) as under Microsoft Windows => start ownCloud client minimized or iconified ;-).

Thanks,

**Comment by lemmy04** [comment]
+1. Same (mis)behaviour on openSUSE Leap 42.2 with Plasma 5.

**Comment by guruz** [comment]
Did you try the 2.3beta1 from https://owncloud.org/changelog/desktop/ ?

**Comment by ldmpub** [comment]
Hi,

I have tried Version 2.3.0rc3 (build 2254) and it seems that this issue has been solved => ownCloud client has been started minimized!

Thanks ;-).

**Comment by guruz** [comment]
Thanks for testing!

**Comment by guruz** [comment]
@ldmpub It would be an idea to try the Qt 5.6.2 based packages:
https://software.opensuse.org/download.html?project=isv%3AownCloud%3Adevel%3AQt562&package=owncloud-client

**Comment by ldmpub** [comment]
@guruz I will try the suggested version tonight but I'm very surprised by your answers: do you want me to install all ownCloud available versions and test them one by one?

**Comment by guruz** [comment]
@ldmpub I'm not surprised by my answer. What else do you want to do?
The linked packages are shipping a newer Qt version which possibly change something there.

**Comment by ldmpub** [comment]
Unfortunately there is no change with Qt 5.6.2 based packages.

Can someone confirm to have seen the following behaviour (I complain not to have): ownCloud started minimized at boot under Linux?

**Comment by guruz** [comment]
I remember this was related to desktop managers restoring the session while ALSO the client is in auto start.
So that "second" launch will tell the client to show the Window (or something like that, i'm an OS X user).

If you remove it from autostart or disable session restore, will you still get this behaviour?

(Could be that app.isSessionRestored is bogus..)

**Comment by ckamm** [comment]
Fwiw: We have a workaround for the "session restore + autostart" issue in the client since before 2.0.2: If the client is run a second time within 10s of the first start, we don't make the settings dialog pop up.

The only other cause I can see would be if `!QSystemTrayIcon::isSystemTrayAvailable() && desktopSession != "ubuntu" && desktopSession != "xfce"`. What's your value of `XDG_CURRENT_DESKTOP` and `DESKTOP_SESSION`?

**Comment by ldmpub** [comment]
@ckamm thanks for your clever answer.

Here are the requested info returned for Xubuntu 16.04 LTS I'm running:
printenv XDG_CURRENT_DESKTOP
XFCE

printenv DESKTOP_SESSION
xubuntu

**Comment by ckamm** [comment]
@ldmpub Thanks! And does your system tray work and the owncloud client pop up in there?

I expect the reason we start with the main window showing for these cases is that we worry that users can't access it otherwise. How does it work in your environment?

**Comment by ldmpub** [comment]
In my environment (Xubuntu 16.04 LTS) the icon in the system tray is 100% operational.

I can either left clic on the minimized icon to open ownCloud main window or right on the icon to open the menu where the "Settings" entry can also open the main ownCloud window like shown below:
![owncloud popup](https://cloud.githubusercontent.com/assets/1452146/24325608/7511ae34-119c-11e7-8234-9218ed51e6da.jpg)

**Comment by ckamm** [comment]
@ldmpub Okay, then you should not be affected by that no-tray-workaround. It also means I don't understand why the settings dialog would pop up for you.

Is this only during startup? What happens when you close the ownCloud client and start it again?

**Comment by ldmpub** [comment]
@ckamm when I close the ownCloud client and start it again everything is fine => the client is started minimized without any popup (it's perfect).

Hope this helps.

**Comment by ckamm** [comment]
@ldmpub Unfortunately it only helps insofar as saying that something is probably up on your end. :/

Either your session initialization starts the client twice, more than 10s apart (you could check whether our double-start workaround works by quitting and then starting it twice in a row, with only a second or so between the two) or maybe your tray becomes available only later? (though I'm not sure that would work)

Currently I don't see what we could do to help without potentially causing other bugs. If you figure out what was going on, please let us know.

**Comment by ldmpub** [comment]
@ckamm it seems that you are right beacause I disabled ownCloud in my "system > applications auto start" and let the option in ownCloud seetings "start with system" enabled and ownCloud is now started minimized on boot.

I need to perform more test. Anyway thank's a lot for your help.

**Comment by ldmpub** [comment]
Hi, after several starts and days used => I confirm everything is ok (see last comment above for solution).

**Comment by lemmy04** [comment]
...not on linux. Running openSUSE Leap 42.2, PLasma 5.6, owncloud-client does NOT start minimized,  no matter how I autostart it (in-app versus desktop settings)

**Comment by guruz** [comment]
@lemmy04 Can you double-check if you are really using the new packages that ship with Qt 5.6.2?
https://github.com/owncloud/client/issues/5528#issuecomment-287038839

**Comment by lemmy04** [comment]
@guruz I'm not... Have to be on QT 5.8.0 for the KDE/Plasma version I'm using...

**Comment by ckamm** [comment]
@lemmy04 Have you tried quitting the client and then restarting it? Does it start in a minimized state this way?

**Comment by lemmy04** [comment]
@ckamm that starts it minimized.

**Comment by guruz** [comment]
@lemmy04 Please use the packages linked above they **ship with their OWN Qt** and re-check.

**Comment by ckamm** [comment]
@lemmy04 Then either you accidentally start owncloud twice during startup more than 10s apart, or the tray area isn't available yet when the client starts. (and neither XDG_CURRENT_DESKTOP nor DESKTOP_SESSION are "ubuntu" or "xfce")

**Comment by lemmy04** [comment]
@guruz I added that repository and installed that version including all deps... same thing.

**Comment by kiview** [comment]
I see the same behavior on Fedora 30 with GNOME 3.32.2, running client `2.5.4 (build 515)`

**Comment by ldmpub** [comment]
Hi @kiview 
Have you checked if the client does not start twice like describe here (workaround for me): https://github.com/owncloud/client/issues/5528#issuecomment-293715744

**Comment by pulkin** [comment]
Same here (Fedora 30, Gnome 3). It either starts windowed or does not start at all, regardless of the combination of startup settings in the app and in Gnome Tweaks. OC v 2.5.4 515.

**Comment by MaxXor** [comment]
Please re-open, issue is still happening on Arch Linux and it's very annoying to close the window each time I'm logging in.

**Comment by AlfTetzlaff** [comment]
The problem persists with Debian 10 / Gnome 3.30.2 and the distro's owncloud version (2.5.1), as no packages for Debian 10 can be found on the owncloud website. No workaround has worked so far. Please reopen this issue.

**Comment by michaelstingl** [comment]
v2.6.0 rc2 has been released for Debian 10. Could you give it a try?
https://software.opensuse.org//download.html?project=isv:ownCloud:desktop:testing&package=owncloud-client

**Comment by AlfTetzlaff** [comment]
v2.6.0 seems to depend on libc6 2.29-2. The current Debian buster version is 2.28-10. I did not dare to update that, as it might break the system (at least my gut-feeling tells me so).

**Comment by AlfTetzlaff** [comment]
I checked with version 2.6.0 rc and the problem persists. I tried turning off the startup in the gnome tweak menu, as suggested earlier, but this really disables the startup and the settings in owncloud and gnome behave coherently (as i would expect).

Furthermore I get other errors: I have two accounts linked to different servers. One of the connections now fails telling me "reading from keychain failed with error: unknown error", the other one always pops up the browser asking me to authorize this app. After entering my pw and authorizing it works - but this is even more annoying.

**Comment by michaelstingl** [comment]
> "reading from keychain failed with error: unknown error"

@AlfTetzlaff maybe better open a new issue for this

**Comment by AlfTetzlaff** [comment]
I found a new workaround. When I install the gnome Topicon plus extension, owncloud doesnt show on startup (nor does any icon in my top bar). I can toggle owncloud's behaviour by switching the extension on and off. Therefore I assume that the original issue is related to gnome having removed the legacy tray. Consequently, there seems to be no convenient way of ending owncloud without opening the list of processes and killing it there.

> > "reading from keychain failed with error: unknown error"
> 
> @AlfTetzlaff maybe better open a new issue for this

I will do so.

**Comment by pehlm** [comment]
Is this going to be resolved? Am using xubuntu 20.04. The client window are showing up at login. There should be some commandline switch to use for minimized window, but not.

**Comment by TheOneRing** [comment]
The client starts minimized by default.
Only if invoked with `-s` or `--showsettings` we will show the ui.
The `-s` parameter is passed by the desktop entry or the Windows shortcut.
